### PR TITLE
💻 Creating Nav Bar

### LINKED ISSUE #63: V0 Navigation
As a student, I want to switch between viewing events and clubs. (the general app navigation bar)

### PR DESCRIPTION
**Pull request by lesliexin**
## Screenshots
**Desktop**
image

**Mobile**
image

## Purpose
Closes #63. As a student, I want to switch between viewing events and clubs. (the general app navigation bar).

## Testing
Testing that the navbar and page options render.

## Learning
Server-side rendering interferes with styled-components. 
To mitigate, we must add the `_document.tsx` file provided by Vercel. More explanation here: https://dev.to/rsanchezp/next-js-and-styled-components-style-loading-issue-3i68

### CODE REVIEW

File: client/styles/index.ts [hunk 1]
```diff
@@ -1,2 +1,3 @@
 export { colours } from "./colours";
 export { device } from "./device";
+export { fontFamily, desktopFontSize, mobileFontSize, fontWeight } from "./typography";
```
Comment (jeffzh4ng) [inline client/styles/index.ts:3]: yay ;)

File: client/components/hooks/UseViewport.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    const [width, setWidth] = useState(0);
+
+    useEffect(() => {
+        const handleWindowResize = () => setWidth(window.innerWidth);
+        window.addEventListener("resize", handleWindowResize);
+        return () => window.removeEventListener("resize", handleWindowResize);
+    }, []);
+
+    return { width };
+}
```
Comment (jeffzh4ng) [inline client/components/hooks/UseViewport.tsx:13]: 😍

File: client/components/__tests__/NavBar.test.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { NavBar } from "../NavBar"
+import { render, screen } from '../../utils/tests/test-utils'
+
+describe('NavBar', () => {
```
Comment (jeffzh4ng) [inline client/components/__tests__/NavBar.test.tsx:4]: love it 😍

**Review by jeffzh4ng — APPROVED** [review]

**Review by jeffzh4ng — APPROVED** [review]

**Comment by amanmathur99** [comment]
Design nit @lesliexin, should there be an active state for the tab you're on? I think the tab colours are inverted rn.

**Review by amanmathur99 — APPROVED** [review]

**Review by jeffzh4ng — APPROVED** [review]

**Review by amanmathur99 — APPROVED** [review]

**Pull request merged**